In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('DATASETS\\uk_road_accident.csv')

In [3]:
df

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h1>1. How many total accidents are in the data?</h1>

In [4]:
df.shape[0]

660679

<h3>Insight : The total number of accidents in the data is 660,679</h3>

<h1>2. How many accidents happened by severity (Slight, Serious, Fatal)?</h1>

In [5]:
df["Accident_Severity"].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>Insight : Here are the total numbers of accidents by severity: Slight - 563,801, Serious - 88,217, and Fatal - 8,661</h3>

<h1>3. What is the most common accident severity?</h1>

In [6]:
df["Accident_Severity"].mode()[0]

'Slight'

<h3>Insight : The most common accident severity is "Slight"</h3>

<h1>4. On average, how many people are hurt in each accident?</h1>

In [7]:
df["Number_of_Casualties"].mean()

np.float64(1.357040257068864)

<h3>Insight : The average number of casualties per accident is 1.35</h3>

<h1>5. What is the highest number of people hurt in a single accident?</h1>

In [8]:
df["Number_of_Casualties"].max()

np.int64(68)

<h3>Insight : The highest number of people hurt in a single accident is 68</h3>

<h1>6. What is the lowest number of people hurt in a single accident?</h1>

In [9]:
df["Number_of_Casualties"].min()

np.int64(1)

<h3>Insight : The lowest number of people hurt in a single accident is 1</h3>

<h1>7. How many total people were hurt in all accidents?</h1>

In [10]:
df["Number_of_Casualties"].sum()

np.int64(896568)

<h3>Insight : The total people were hurt in all accidents is 896,568</h3>

<h1>8. On average, how many vehicles are involved in one accident?</h1>

In [11]:
df["Number_of_Vehicles"].mean()

np.float64(1.8312554205597575)

<h3>Insight : The average vehicles per accident is 1.83</h3>

<h1>9. What is the highest number of vehicles in one accident?</h1>

In [12]:
df["Number_of_Vehicles"].max()

np.int64(32)

<h3>Insight : The highest number of vehicles in one accident is 32</h3>

<h1>10. What is the most common number of vehicles in an accident?</h1>

In [13]:
df["Number_of_Vehicles"].mode()[0]

np.int64(2)

<h3>Insight : The most common number of vehicles in an accident is 2</h3>

<h1>11. How many accidents happened under each light condition?</h1>

In [14]:
df["Light_Conditions"].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h3>Insight : Here are the accidents happened under each light condition:<br>Daylight - 484,880<br>Darkness(lights lit) - 129,335<br>Darkness(no lighting) - 37,437<br>Darkness(lighting unknown) - 6,484<br>Darkness(lights unlit) - 2,543
</h3>

<h1>12. What is the most common light condition during accidents?</h1>

In [15]:
df["Light_Conditions"].mode()[0]

'Daylight'

<h3>Insight : The most common light condition during accidents is Daylight</h3>

<h1>13. How many accidents happened under each weather condition?</h1>

In [16]:
df["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insight : Here are the accidents happened under each weather condition:<br>Fine no high winds - 520,885<br>Raining no high winds - 79,696<br>Other - 17,150<br>Raining + high winds - 9,615<br>Fine + high winds) - 8,554<br>Snowing no high winds - 6,238<br>Fog or mist - 3,528<br>Snowing + high winds - 8,85
</h3>

<h1>14. What is the most common weather condition during accidents?</h1>

In [17]:
df["Weather_Conditions"].mode()[0]

'Fine no high winds'

<h3>Insight : The most common weather condition during accidents is 'Fine no high winds'
</h3>

<h1>15. Which 5 districts have the most accidents?</h1>

In [18]:
df["District Area"].value_counts().head(5)

District Area
Birmingham    13491
Leeds          8898
Manchester     6720
Bradford       6212
Sheffield      5710
Name: count, dtype: int64

<h3>Insight : The 5 districts that have the most accidents are:<br>Birmingham - 13,491<br>Leeds - 8,898<br>Manchester - 6,720<br>Bradford - 6,212<br>Sheffield - 5,710</h3>

<h1>16. How many accidents happened on each road surface condition?</h1>

In [19]:
df["Road_Surface_Conditions"].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>Insight : Here are the accidents that happened on each road surface condition:<br>Dry - 447,821<br>Wet or damp - 186,708<br>Frost or ice - 18,517<br>Snow - 5,890<br>Flood over 3cm. deep - 1,017</h3>

<h1>17. What is the most common road surface condition in accidents?</h1>

In [20]:
df["Road_Surface_Conditions"].mode()[0]

'Dry'

<h3>Insight : The most common road surface condition in accidents is 'Dry'
</h3>

<h1>18. How many accidents happened in urban vs rural areas?</h1>

In [21]:
df["Urban_or_Rural_Area"].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h3>Insight : The number of accidents happened in urban vs rural areas are<br>Urban - 421,663<br>Rural - 238,990<br>Unallocated - 11</h3>

<h1>19. Which 5 vehicle types are most often in accidents?</h1>

In [22]:
df["Vehicle_Type"].value_counts().head(5)

Vehicle_Type
Car                                     497992
Van / Goods 3.5 tonnes mgw or under      34160
Bus or coach (17 or more pass seats)     25878
Motorcycle over 500cc                    25657
Goods 7.5 tonnes mgw and over            17307
Name: count, dtype: int64

<h3>Insight : Here are the 5 vehicle types that are most often in accidents:<br>Car - 497,992<br>Van / Goods 3.5 tonnes mgw or under - 34,160<br>Bus or coach (17 or more pass seats) - 25,878<br>Motorcycle over 500cc - 25,657<br>Goods 7.5 tonnes mgw and over - 17,307</h3>

<h1>20. What are the summary stats for casualties and vehicles?</h1>

In [23]:
df[["Number_of_Casualties", "Number_of_Vehicles"]].describe()

Number_of_Casualties  Number_of_Vehicles
count         660679.000000       660679.000000
mean               1.357040            1.831255
std                0.824847            0.715269
min                1.000000            1.000000
25%                1.000000            1.000000
50%                1.000000            2.000000
75%                1.000000            2.000000
max               68.000000           32.000000

<h3>Insight : This shows the smallest, biggest, and average numbers for both casualties and vehicles in the accidents.</h3>